languageServer/src/languageHelpers/baseLanguageHelper.ts

```diff
@@ -0,0 +1,105 @@
+import { has } from 'lodash'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+  DocumentSymbol,
+  Position,
+  Location,
+  SymbolKind
+} from 'vscode-languageserver/node'
+
+export interface LanguageHelper {
+  findSymbolAtPosition(position: Position): DocumentSymbol | undefined
+  findLocationsFor(symbol: DocumentSymbol): Location[]
+}
+
+export abstract class BaseLanguageHelper<RootNode> implements LanguageHelper {
```

Comment:
It's usual practice in the codebase to have this a `<T>`. I was looking for where `<RootNode>` is defined for a couple of minutes.